open_spiel/games/dark_hex.h

```diff
@@ -0,0 +1,152 @@
+// Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef OPEN_SPIEL_GAMES_DARK_HEX_H_
+#define OPEN_SPIEL_GAMES_DARK_HEX_H_
+
+#include <array>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "open_spiel/games/hex.h"
+#include "open_spiel/spiel.h"
+
+// Dark Hex (Some versions also called Phantom Hex or Kriegspiel Hex) is an imperfect
+// information version of the classic game of Hex. Players are not exposed to oppsite
+// sides piece information. Only a refree has the full information of the board. When
+// a move fails due to collision the player gets the information of the cell (stone
+// exists), and is allowed to make another move until success. 
+// 
+// There are slightly different versions of the game exists depending on the level of
+// information being exposed to the opponent and what happens in the event of a
+// collision. Right now we only have the Classic Dark Hex (Phantom Hex) implemented 
+// with player:
+//        - Replays after a collision
+//        - Exposed(information on opponent) only to number of turns or nothing
```

Comment:
Can we clarify this?

Suppose we have a sequence:
- P0 plays position X
- P1 plays position Y
- P0 plays position Y (collision?)
- P0 plays position Z
- P1's turn

What information does P1 have? Do they
(a) know there's been a total of 4 turns taken (hence that P0 collided)
(b) only know that they have taken 1 turn so far (hence P0 played 2 valid moves, but does not know the number of collisions)
(c) something else?

Reply:
Ahh, the implementation supports both via Observation type. I see. Can you clarify that here in the comment and mention the obs_type game parameter?